### ISSUE TITLE
New way to create nodes

### ISSUE THREAD

**Issue by ai** [postcss/postcss#502]
Right now some plugin create nodes manually:

``` js
let decl = postcss.decl({ prop: 'color', value: 'black' });
```

But on major release, plugin’s `postcss` can be different (for example, 4.1 version) rather than runner `postcss` (for example, 5.0). As result `postcss.decl` will generate old instance with old API.

AST inconsistency is very dangerous thing. Because of npm tree nature, it can happen even on non-major updates.

So, I think we need to depreacte `postcss.*` creation methods and suggest different way.

**Comment by ai** [comment]
First long-term solution suggestion:

```
let decl = css.create('atrule', { name: 'encoding' })
```

**Comment by ai** [comment]
Second solution is to send `postcss` instance to plugin function. It is better, because some plugins uses `postcss.parse`.

Maybe also we should move `postcss.plugin` to other project to avoid use old PostCSS API:

```js
import plugin from `postcss-plugin`;

export default plugin( (css, result, postcss) => {
});
```

**Comment by ai** [comment]
@lydell @MoOx what do you think? You had a great ideas about API.

**Comment by lydell** [comment]
Not having thought it through _that_ much yet, but I think I like your last idea.

**Comment by ai** [comment]
Dirty fix will be released in 5.0.1: https://github.com/postcss/postcss/commit/e7f2d0f955425fe4769adfae17309ca0768ca066

**Comment by 1j01** [comment]
Having the `postcss` instance available to the plugin could also be useful for things in the future like [maybe `postcsss.func()`](https://github.com/postcss/postcss/issues/178)
